be/src/io/cache/cache_lru_dumper.cpp

```diff
@@ -0,0 +1,375 @@
+    // +-----------------------------------------------+
+    // | LRUDumpEntryGroupPb_3                         |
+    // +-----------------------------------------------+
+    // | ...                                           |
+    // +-----------------------------------------------+
+    // | LRUDumpEntryGroupPb_n                         |
+    // +-----------------------------------------------+
+    // | LRUDumpMetaPb (queueName, Group<off,size>List)|
+    // +-----------------------------------------------+
+    // | FOOTER_OFFSET (8Bytes)                        |
+    // +-----------------------------------------------+
+    // | CHECKSUM (4Bytes)｜VERSION (1Byte)｜ MAGIC     |
+    // +-----------------------------------------------+
+    //
+    // why we are not using protobuf as a whole?
+    // AFAIK, current protobuf version dose not support streaming mode,
+    // so that we need to store all the message in memory which will
+    // consume loads of RAMs.
+    // Instead, we use protobuf serialize each of the single entry
+    // and provide the version field in the footer for upgrade
+
+    ::doris::io::cache::LRUDumpEntryPb* entry = _current_dump_group.add_entries();
+    ::doris::io::cache::UInt128WrapperPb* hash_pb = entry->mutable_hash();
+    hash_pb->set_high(hash.high());
+    hash_pb->set_low(hash.low());
+    entry->set_offset(offset);
+    entry->set_size(size);
+
+    _current_dump_group_count++;
+    if (_current_dump_group_count >= 10000) {
+        RETURN_IF_ERROR(flush_current_group(out, filename));
+    }
+    return Status::OK();
+}
+
+Status CacheLRUDumper::flush_current_group(std::ofstream& out, std::string& filename) {
+    if (_current_dump_group_count == 0) {
+        return Status::OK();
+    }
+
+    // Record current position as group start offset
+    size_t group_start = out.tellp();
+
+    // Serialize and write the group
+    std::string serialized;
+    LOG(INFO) << "Before serialization: " << _current_dump_group.DebugString();
+    if (!_current_dump_group.SerializeToString(&serialized)) {
+        return Status::InternalError<false>("Failed to serialize LRUDumpEntryGroupPb");
+    }
+    LOG(INFO) << "Serialized size: " << serialized.size();
+    out.write(serialized.data(), serialized.size());
+    RETURN_IF_ERROR(check_ofstream_status(out, filename));
+
+    // Record group metadata
+    ::doris::io::cache::EntryGroupOffsetSizePb* group_info = _dump_meta.add_group_offset_size();
+    group_info->set_offset(group_start);
+    group_info->set_size(serialized.size());
+
+    // Reset for next group
+    _current_dump_group.Clear();
+    _current_dump_group_count = 0;
+    return Status::OK();
+}
+
+Status CacheLRUDumper::finalize_dump(std::ofstream& out, size_t entry_num,
+                                     std::string& tmp_filename, std::string& final_filename,
+                                     size_t& file_size) {
+    // Flush any remaining entries
```

Comment:
improve observibility and more stats e.g. log total number of entries, size in bytes flushed, latency and etc.
also consider export some bvars metrics to make it easier to access